### PR TITLE
Language Authoring 2022-10-01-preview

### PR DESCRIPTION
**Pull request by AmirGeorge**
Data Plane API - Pull Request
Adding Language Authoring 2022-10-01-preview APIs to the specification folder

API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [engagement record](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

Link to engagement record issue:
Is this review for (select one):

a public preview

Change Scope
This section will help us focus on the specific parts of your API that are new or have been modified.
Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated.

Design Document:
Previous Open API Doc:
Updated paths:
❔Got questions? Need additional info?? We are here to help!
Contact us!
Click here for links to tools, specs, guidelines & other good stuff

### CODE REVIEW

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:670]: To aid our code generators, please tell us how to get the final value. See https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-long-running-operation-options

Or is that inline on the job status described below?

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:2003]: Can you describe what regex syntax is required e.g., .NET, Perl, etc? There are so many dialects that differ, I think it's important say at least what dialect is supported here, ideally with a link.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:2522]: Same comment here (or anywhere you mention regex): what dialect is supported?

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:2975]: If this is only an extension of the generic (authoring) job state, how does one get the final value? See https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#long-running-operations--jobs for our current standards.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext-authoring.json:1538]: What creates a type of job that uses this for a status monitor? I don't see anything obvious in this swagger.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext-authoring.json:3175]: I recommend document this is between 0 and 1.

**Review by heaths — CHANGES_REQUESTED** [review]
Mostly minor things and a question, but copy/paste my comments on analyzeconversations-authoring.json to analyzetext-authoring.json where appropriate since they seem very close to duplicate authoring changes.

Also, once this is merged, you, @kayousef, or someone will want to robocopy all the changes made to these files under _specification/.../2022-10-01-preview_ to the _dev/_ folder just to make sure all major additions like this and minor tweaks for PRs are contained therein. _dev/_ should always be the latest in-development copy (like you'd typically do with source code) and be an evolution of the last published preview or stable API.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (AmirGeorge) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:2003]: Will ask the developer who owns regex for clarification on this.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (AmirGeorge) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:670]: Yes it is inline in the job status. There is no "result" for this operation, the user would only need to monitor the status till it is success.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (AmirGeorge) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:2975]: Same as the other comment, the final value is the inline status when it reaches a terminal state.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext-authoring.json
Comment (AmirGeorge) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzetext-authoring.json:1538]: That would be the delete project API. Abbas who was on my team has asked the board about this before. The issues he faced was that the get status for the delete project operation couldn't projects/{projectName}/jobs/{jobId} because as soon as the project is actually deleted, then the get status would return a 404. So the board recommendation was to use this for create project get status.

This is the same as what we have in our 2022-05-01 GA APIs by the way, the signature hasn't changed.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (AmirGeorge) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:670]: This is the same format we used have been using in our previous API versions including the GA 2022-05-01, it was discussed with the board back when Abbas was working on the APIs.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:670]: But in most other LROs, we typically get the final object from the original URI. Is there any final object in these cases? How should clients get it? It's not obvious in `:verb`-style LROs.

Comment (AmirGeorge) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:670]: @heaths as communicated in the chat, In most of the :verb style LROs there is no final object. The one in this comment thread for example deletes a deployment, so there is no final object. If for an LRO there is a final object, whether :verb style or not, there is the result url property in the get status response, like what happens in :export:
image

Also as mentioned in the other comment, this is not a new pattern being introduced in 2022-10-01-preview. This is the same LRO design we have been following in all the previous API versions including the GA 2022-05-01, so if there is room for improvement here I suggest we take the discussion for another time and not block merging the PR on it. Thoughts?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:670]: We can discuss separately, but note that what you don't author here for a final object (and there is often for POST-style LROs) we actually end up having to add for code generation. See https://github.com/Azure/azure-sdk-for-net/blob/48b37c8fb68e9446f78b0c014169b2e065e1806a/sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/autorest.md?plain=1#L67-L75

/cc @kristapratico @mayurid

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:2526]: Do not include the locale in URLs as this forces everyone to en-US instead of using their browser to decide:
```suggestion
          "description": "The regex expressions of the regex component. These expressions follow the .NET regex syntax. For reference, check here: https://learn.microsoft.com/dotnet/standard/base-types/regular-expressions",
```

Same for any URL below.

File: specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-10-01-preview/analyzeconversations-authoring.json:2536]:
```suggestion
      "description": "Represents a regex expression inside a regex component. This expression follows the .NET regex syntax. For reference, check here: https://learn.microsoft.com/dotnet/standard/base-types/regular-expressions",
```

**Review by heaths — CHANGES_REQUESTED** [review]

**Review by heaths — APPROVED** [review]

**Comment by heaths** [comment]
@lmazuel can you override and merge? This is a false hit since the different swaggers generate different clients. Back in the _dev/_ folder in `main` I could fix this to at least prevent this from erring in the future, but for now can you help unblock this change?

```
"location":"specification/cognitiveservices/data-plane/Language/readme.md",
"tag":"[release_2022_10_01_preview](https://github.com/Azure/azure-rest-api-specs/blob/9b5852c414e04fe51c97c3a40820df937d1d937e/specification/cognitiveservices/data-plane/Language/readme.md#tag-release_2022_10_01_preview)",
"details":"Error: '$["x-ms-parameterized-host"].hostTemplate' has incompatible values (---'Endpoint/language',
---'Endpoint/language/authoring')."
```

**Comment by weshaggard** [comment]
@heaths please either file an issue to fix the authoring issue or the tooling for this.

**Pull request merged**
